tests/integration/conftest.py

```diff
@@ -109,6 +115,136 @@ def factory(command, silent=False, hide_command=False):
+
+        return
+
+    def install_package(self, package_name="subscription-manager"):
+        """
+        Install a package.
+        :param package_name: The package to be installed. Default is subscription-manager.
+        :type package_name: str
+        """
+        command = f"yum install -y {package_name}"
+        # rhn-client-tools package obsoletes subscription-manager on Oracle Linux
+        # set the obsoletes option to 0 to be able to install the package
+        if SystemInformationRelease.distribution == "oracle":
+            command += " --setopt=obsoletes=0"
+
+        return self.shell(command)
+
+    def remove_package(self, package_name="subscription-manager*"):
+        """
+        Removes a package.
+        :param package_name: The package to be installed. Default is subscription-manager*.
+        :type package_name: str
+        """
+        command = f"yum remove -y {package_name}"
+        return self.shell(command)
+
+    def remove_client_tools_repo(self):
+        """
+        Remove the client tools repository file.
+        """
+        command = "rm -f /etc/yum.repos.d/client-tools-for-rhel*.repo"
+
+        return self.shell(command)
+
+    def remove_keys_and_certificates(self):
+        """
+        Remove the SSL certificate for accessing the CDN and the redhat RPM GPG key.
+        """
+        self.shell("rm -f /etc/rhsm/ca/redhat-uep.pem")
+        self.shell("rm -f /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release")
+
+        return
+
+    def unregister(self):
+        """
+        Remove potential leftover subscription, unregister the system.
+        """
+        # Remove potential leftover subscription
+        self.shell("subscription-manager remove --all")
+        # Remove potential leftover registration
+        self.shell("subscription-manager unregister")
+
+        return
+
+    def prepare_full_workflow(self):
+        """
+        Usual full preparation workflow.
+        Calls, where applicable:
+            # only on Oracle Linux
+            self.remove_package(package_name="rhn-client-tools")
+            self.add_keys_and_certificates()
+            self.add_client_tools_repo()
+            self.install_package()
+        """
+        if SystemInformationRelease.distribution == "oracle":
+            self.remove_package(package_name="rhn-client-tools")
+        self.add_keys_and_certificates()
+        self.add_client_tools_repo()
+        self.install_package()
+
+        return
+
+    def teardown_full_workflow(self):
+        """
+        Usual full teardown workflow.
+        Calls where applicable:
+            self.unregister()
+            self.remove_package()
+            self.remove_client_tools_repo()
+            self.remove_keys_and_certificates()
+        """
+        self.unregister()
+        self.remove_package()
+        self.remove_client_tools_repo()
+        self.remove_keys_and_certificates()
+
+        return
+
+
+SUBMAN = SubscriptionManager()
+
```

Comment:
Is this needed? I would try to avoid it if possible. If some fixture or function needs it, then it can instantiate its own

Reply:
Makes sense